dask/dataframe/io/parquet/core.py

```diff
@@ -556,45 +556,60 @@ def apply_filters(parts, statistics, filters):
         Tokens corresponding to row groups to read in the future
     statistics: List[dict]
         List of statistics for each part, including min and max values
-    filters: List[Tuple[str, str, Any]]
-        List like [('x', '>', 5), ('y', '==', 'Alice')]
+    filters: Union[List[Tuple[str, str, Any]], List[List[Tuple[str, str, Any]]]]
+        List of filters to apply, like [[('x', '=', 0), ...], ...]
 
```

Comment:
Why the `"Union"` here?

Also, you are right that the documentation is lacking here, and the convention for defining filters is not intuitive at first glance.  Maybe we can copy some of the same information used in [the pyarrow documentation](https://arrow.apache.org/docs/python/generated/pyarrow.parquet.ParquetDataset.html).  For example, perhaps we can add something like:

> This implements partition-level (hive) filtering only, i.e., to prevent the loading of some row-groups and/or files.

> Predicates can be expressed in disjunctive normal form (DNF). This means that the innermost tuple describes a single column predicate. The list of inner predicates are combined with an AND conjunction into a larger predicate. The outer-most list then combines all of the combined filters with an OR disjunction.

Reply:
Certainly second that there should be examples of the possible types of input here

Reply:
Thanks for reviewing! I've now added the documentation to `apply_filters` and `read_parquet`.

> Why the `"Union"` here?

The `List[Tuple]` case represents a single conjunction (the old behaviour), and `List[List[Tuple]]` represents disjunctive normal form. This is also how PyArrow handles it.

Reply:
The added documentation looks good.  Although, I'm not sure if we should add it to `read_parquet` unless the same options apply to both pyarrow and fastparquet

Reply:
Is there a reason this won't apply to fastparquet? Is the logic just being passed on to pyarrow, then?

Reply:
The logic should not depend on the underlying engine, as long as the min/max statistics are available.  You are right that this should apply to fastparquet, and so we should be testing it.